lib/src/chewie_player.dart

```diff
@@ -129,6 +129,7 @@ class ChewieState extends State<Chewie> {
     SystemChrome.setEnabledSystemUIOverlays([]);
     if (isAndroid) {
       SystemChrome.setPreferredOrientations([
+        DeviceOrientation.portraitUp,
```

Comment:
Is this required for the fix as well? If so, I think it might be good to completely remove these preferredOrientations as the only one missing would be portraitDown.

Reply:
Good point! i'll remove it and test it

Reply:
Do we need this:
https://github.com/brianegan/chewie/blob/02b3bf2f9d3bdb669c1bec92dcd5bac0ea508e03/lib/src/chewie_player.dart#L150
and this:
https://github.com/brianegan/chewie/blob/02b3bf2f9d3bdb669c1bec92dcd5bac0ea508e03/lib/src/chewie_player.dart#L152
anymore, if we remove PreferredOrientations and EnabledSystemUIOverlays?

Reply:
I think we'll definitely want to keep the `setEnabledSystemUIOverlays `, but the preferred orientations is trickier.

I could see taking a few approaches, would love input from you and others on what would work best for everyone:

  1. Keep the orientation lock, but make it smarter:  set the preferred orientations based on the aspect ratio of the video, e.g. PortraitUp/Down for h > w videos, and landscapeLeft/Right for w > height videos. This would still allow the "auto switch to landscape in fullscreen" which is common for Android devices, but would still work for portrait vids. This means we would keep the `setPreferredOrientations` line from above when exiting fullscreen.
  2. Allow folks to customize those orientations, as is done in this PR:. Also keeps the `setPreferredOrientations` line from above when exiting fullscreen.
  3. A combination of those two: Use the orientations defined by the user if provided, fallback to detecting the appropriate orientation lock if not. Also keeps the `setPreferredOrientations` line from above when exiting fullscreen.
  4. Remove those lines! The only downside: That would disable the "auto rotate to Landscape upon entering fullscreen" feature, which feels like the default for most android video players. In this case, we could remove the `setPreferredOrientations` line when exiting fullscreen.

Do you have an opinion on which of these three would be the best option for your use case? I only use Chewie for a few small things, so I'd love your feedback!

Reply:
Thank you again @brianegan for your time to discuss about the solution for this. 

I will summarize it here in short:

1. We'll have two new parameters, that users can set and have default values:
```dart
    this.systemOverlaysOnEnterFullScreen = SystemUiOverlay.values,
    this.deviceOrientationsOnEnterFullScreen = DeviceOrientation.values,
``` 
2. Chewie will set the `setPreferredOrientations` based on the videos aspectRatio (h>w = Portrait and w>h = Landscape) for both use cases: OnEnterFullScreen and AfterFullScreen. Is that correct? Or just for OnEnterFullScreen?

Thanks for your feedback :)